torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1241,37 +1334,68 @@ Val* hoistProducerIndex(
     const IndexCompute& ref_indexing,
     const std::vector<kir::ForLoop*>& loops,
     Val* index) {
-  // If index has no defining expression, there's nothing to hoist
-  if (isDisabled(DisableOption::IndexHoist) || index->definition() == nullptr) {
+  auto maybe_indexed_producer_id = getMaybeIndexedIdToHoist(
+      producer_root_id, producer_tv, producer_indexing, index);
+
+  if (!maybe_indexed_producer_id.has_value()) {
     return index;
   }
 
-  // The old swizzle interface, which should be deprecated, is not
-  // supported.
-  if (producer_tv->swizzleType() != SwizzleType::NoSwizzle) {
+  auto indexed_consumer_id_it = p2c_map.find(maybe_indexed_producer_id.value());
+
+  // There can be no corresponding consumer ID. For example, consider:
+  //   consumer: [b1, i2, i3]
+  //   producer: [i2, i3].
+  // Suppose the consumer is transformed as:
+  //   consumer: [(b1*i2)*i3]
+  // Then the producer would be transformed when indexed:
+  //   producer: [i2*i3]
+  // Assuming i2 and i3 are contiguous, the producer indexing is done
+  // with the mreged i2*i3 domain, but there's no domain in the
+  // cosumer that maps with the producer indexed domain.
+  // It seems non-trivial to support patterns like this. Skip for now.
```

Comment:
Yeah, I agree. I think this pattern isn't important to handle any time soon. More important I think we need to change what we consider for hoisting, as right now we only consider hoisting the final indexing at the root. Seems we would want a progressive hoisting where we check hoisting indexing closer to the leaf domains.

Reply:
I don't see value here, but actually couldn't we just use best effort replay to ignore the broadcast axes in consumer? If we can map across while ignoring broadcast which we should be able to here, we should be able to hoist.

Reply:
Oh, just realized this comment was copy-pasted 😆

Reply:
We might have a chance at fixing this at later stage of refactoring. Will revisit.